### ISSUE TITLE
Check statically known computed names in no-dupe-class-members

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12808]
**What rule do you want to change?**

`no-dupe-class-members`

**Does this change cause the rule to produce more or fewer warnings?**

more

**How will the change be implemented? (New option, new default behavior, etc.)?**

new default behavior

Maybe as a breaking change in `v7.0.0`?

**Please provide some example code that this change will affect:**

```js
class A {
  ["a"](){}
  ["a"](){}
  
  [`b`](){}
  b(){}
}
```

**What does the rule currently do for this code?**

no errors

**What will the rule do after it's changed?**

2 errors, for names `"a"` and `"b"`.

The proposal is to check all class members for which `astUtils.getStaticPropertyName` returns a string value and thus align this rule to the similar `no-dupe-keys` rule:

```js
/*eslint no-dupe-keys: "error"*/
/*eslint no-dupe-class-members: "error"*/

var obj =  {  
  ["a"](){},
  ["a"](){}, // error
  
  [`b`](){},
  b(){} // error
}

class A {
  ["a"](){}
  ["a"](){} // should be error
  
  [`b`](){}
  b(){} // should be error
}
```

[Demo Link](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tZHVwZS1rZXlzOiBcImVycm9yXCIqL1xuLyplc2xpbnQgbm8tZHVwZS1jbGFzcy1tZW1iZXJzOiBcImVycm9yXCIqL1xuXG52YXIgb2JqID0gIHsgIFxuICBbXCJhXCJdKCl7fSxcbiAgW1wiYVwiXSgpe30sIC8vIGVycm9yXG4gIFxuICBbYGJgXSgpe30sXG4gIGIoKXt9IC8vIGVycm9yXG59XG5cbmNsYXNzIEEge1xuICBbXCJhXCJdKCl7fVxuICBbXCJhXCJdKCl7fSAvLyBzaG91bGQgYmUgZXJyb3JcbiAgXG4gIFtgYmBdKCl7fVxuICBiKCl7fSAvLyBzaG91bGQgYmUgZXJyb3Jcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

**Are you willing to submit a pull request to implement this change?**

Yes.

**Comment by kaicataldo** [comment]
Agreed. I think this should be treated as a breaking bug fix, since we're currently release v7.

**Comment by mdjermanovic** [comment]
I'm working on this.
